coalib/settings/ConfigurationGathering.py

```diff
@@ -385,6 +386,19 @@ def gather_configuration(acquire_settings,
     local_bears, global_bears = fill_settings(sections,
                                               acquire_settings,
                                               log_printer)
+
+    arg_parser = default_arg_parser() if arg_parser is None else arg_parser
```

Comment:
@Nosferatul What is your opinion about moving this whole deleting section into `load_configuration()` https://github.com/coala/coala/blob/master/coalib/settings/ConfigurationGathering.py#L139?

IMO it's better placed here because that function is intended to "parses the CLI args and loads the config file accordingly".